<a href="https://colab.research.google.com/github/jnrxvr/python-curso-em-video/blob/main/Python_Mundo3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# NOTEBOOK DOS EXERCÍCIOS DO MUNDO 3 DE PYTHON DO CURSO EM VÍDEO

# TUPLAS

## Exercício Python 072: Crie um programa que tenha uma dupla totalmente preenchida com uma contagem por extenso, de zero até vinte. Seu programa deverá ler um número pelo teclado (entre 0 e 20) e mostrá-lo por extenso.

In [ ]:
# entrada de dados pelo usuário

num = int(input('Digite um número inteiro de 0 a 20 para saber seu nome por extenso: '))

# declaração das variáveis

tupla = ('zero', 'um', 'dois', 'três', 'quatro', 'cinco', 'seis', 'sete', 'oito', 'nove', 'dez',
         'onze', 'doze', 'treze', 'catorze', 'quinze', 'dezesseis', 'dezessete', 'dezoito', 'dezenove', 'vinte')

# exibição dos resultado

print(f'\nO número digitado foi: {num}'
      f'\nSeu valor por extenso é : {tupla[num]}')



Digite um número inteiro de 0 a 20 para saber seu nome por extenso: 20

O número digitado foi: 20
Seu valor por extenso é : vinte


## Exercício Python 073: Crie uma tupla preenchida com os 20 primeiros colocados da Tabela do Campeonato Brasileiro de Futebol (utilize tabela de 2023), na ordem de colocação. Depois mostre:

1. Os 5 primeiros times.
2. Os últimos 4 colocados.
3. Times em ordem alfabética.
4. Em que posição está o time do CRUZEIRO

In [ ]:
""" note que na tupla abaixo os times estão em ordem de colocação no brasileirão 2023,
porém os índices não correspondem com a posição correta, pois não existe posição 0, ou seja não existe um time que ficou em 0º lugar """

# declaração das variáveis

brasileirao = ('PALMEIRAS', 'GRÊMIO', 'ATLÉTICO-MG', 'FLAMENGO', 'BOTAFOGO', 'BRAGANTINO', 'FLUMINENSE', 'ATHLÉTICO-PR', 'INTERNACIONAL', 'FORTALEZA',
               'SÃO PAULO', 'CUIABÁ', 'CORINTHIANS', 'CRUZEIRO', 'VASCO DA GAMA', 'BAHIA', 'SANTOS', 'GOIÁS', 'CORITIBA', 'AMÉRICA-MG')

print(f'Os 5 primeiros colocados são: {brasileirao[0:5]}')
print(f'Os 4 últimos colocados são: {brasileirao[16:]}') # note que o SANTOS ficou em 17º lugar, porém seu índice é 16, pois em python os índices começam em 0
print(f'Os times colocados em ordem alfabética: {sorted(brasileirao)}')
print(f"A posição do CRUZEIRO na tabela é: {brasileirao.index('CRUZEIRO')+1}") # foi somado +1, pois os índices começam em 0 e não refletem as posições corretamente

Os 5 primeiros colocados são: ('PALMEIRAS', 'GRÊMIO', 'ATLÉTICO-MG', 'FLAMENGO', 'BOTAFOGO')
Os 4 últimos colocados são: ('SANTOS', 'GOIÁS', 'CORITIBA', 'AMÉRICA-MG')
Os times colocados em ordem alfabética: ['AMÉRICA-MG', 'ATHLÉTICO-PR', 'ATLÉTICO-MG', 'BAHIA', 'BOTAFOGO', 'BRAGANTINO', 'CORINTHIANS', 'CORITIBA', 'CRUZEIRO', 'CUIABÁ', 'FLAMENGO', 'FLUMINENSE', 'FORTALEZA', 'GOIÁS', 'GRÊMIO', 'INTERNACIONAL', 'PALMEIRAS', 'SANTOS', 'SÃO PAULO', 'VASCO DA GAMA']
A posição do CRUZEIRO na tabela é: 14


## Exercício Python 074: Crie um programa que vai gerar cinco números aleatórios e colocar em uma tupla. Depois disso, mostre a listagem de números gerados e também indique o menor e o maior valor que estão na tupla.

In [ ]:
# importando a biblioteca random
from random import randint

# gerando os números pseudoaleatórios

n1 = randint(1,10)
n2 = randint(1,10)
n3 = randint(1,10)
n4 = randint(1,10)
n5 = randint(1,10)

# declarando a tupla com os números gerados
tuplaNum = (n1, n2, n3, n4, n5)
print(tuplaNum) # exibindo os números sorteados

# apresentação dos resultados

print(f'O maior valor da tupla é: {max(tuplaNum)}')
print(f'O menor valor da tupla é: {min(tuplaNum)}')

O maior valor da tupla é: 10
O menor valor da tupla é: 2


In [ ]:
# SEGUNDA FORMA DE RESOLVER O EXERCÍCIO

# importando a biblioteca random
from random import randint

# declarando a tupla com os números gerados
tuplaNum = (randint(1,10), randint(1,10), randint(1,10), randint(1,10), randint(1,10))

for n in tuplaNum:
    print(f'{n}', end=' ') # exibindo os números sorteados
# apresentação dos resultados

print(f'\nO maior valor da tupla é: {max(tuplaNum)}')
print(f'O menor valor da tupla é: {min(tuplaNum)}')

3 7 1 1 8 
O maior valor da tupla é: 8
O menor valor da tupla é: 1


## Exercício Python 075: Desenvolva um programa que leia quatro valores pelo teclado e guarde-os em uma tupla. No final, mostre:

1. Quantas vezes apareceu o valor 9.
2. Em que posição foi digitado o primeiro valor 3.
3. Quais foram os números pares.

In [ ]:
# entrada de dados pelo usuário

n1 = float(input('Digite o primeiro valor: '))
n2 = float(input('Digite o segundo valor: '))
n3 = float(input('Digite o terceiro valor: '))
n4 = float(input('Digite o quarto valor: '))

# adicionando em uma tupla, os números digitados pelo usuário

numeros = (n1, n2, n3, n4)

# apresentando os resultados

print(f'O número 9 apareceu {numeros.count(9)} vez(es).') # conta quantas vezes um elemento apareceu na tupla

"""quando o python não encontra um valor com o método 'count', ele apresenta uma mensagem de erro (tuple.index(x): x not in tuple),
para resolver esse problema, foi utilizado a estrutura if abaixo: """

if 3 in numeros:
    print(f'A posição em que foi digitado o primeiro valor 3 é: {numeros.index(3)}.') # indica o índice da primeira ocorrência do elemento desejado
else:
   print('O valor 3 não foi digitado em nenhuma posição!')

print('Os números pares digitados foram: ', end='')
for n in numeros:
    if n % 2 == 0:
        print(f'{n}', end=' ')


Digite o primeiro valor: 1
Digite o segundo valor: 2
Digite o terceiro valor: 4
Digite o quarto valor: 5
O número 9 apareceu 0 vez(es).
O valor 3 não foi digitado em nenhuma posição!
Os números pares digitados foram: 2.0 4.0 

In [ ]:
# SEGUNDA FORMA DE RESOLVER O PROBLEMA

# entrada de dados pelo usuário

n1 = float(input('Digite o primeiro valor: '))
n2 = float(input('Digite o segundo valor: '))
n3 = float(input('Digite o terceiro valor: '))
n4 = float(input('Digite o quarto valor: '))

# adicionando em uma tupla, os números digitados pelo usuário

numeros = (n1, n2, n3, n4)


# apresentando os resultados

print(f'O número 9 apareceu {numeros.count(9)} vezes.') # conta quantas vezes um elemento apareceu na tupla
print(f'O índice da primeira ocorrência do valor 3 é: {numeros.index(3)}.') # indica o índice da primeira ocorrência do elemento desejado
print(f'Os números pares digitados foram: {[n for n in numeros if n % 2 == 0]}') # forma reduzida do if, para caber direto no print

Digite o primeiro valor: 2
Digite o segundo valor: 4
Digite o terceiro valor: 9
Digite o quarto valor: 3
O número 9 apareceu 1 vezes.
O índice da primeira ocorrência do valor 3 é: 3.
Os números pares digitados foram: [2.0, 4.0]


## Exercício Python 076: Crie um programa que tenha uma tupla única com nomes de produtos e seus respectivos preços, na sequência. No final, mostre uma listagem de preços, organizando os dados em forma tabular.

In [ ]:
# criando a tupla com os nomes de produtos e seus respectivos preços

produtos = ('Arroz', 5.53, 'Feijão', 4.32, 'Macarrão', 3.25, 'Farinha', 6.47, 'Carne', 17.83, 'Frango', 9.67,)


# nesse caso, temos que usar o for com range para iterar sobre a tupla

print('-------------------------TABELA DE PREÇOS-------------------------')
for c in range(0, len(produtos), 2):
    print(f'\n{produtos[c]:.<58} R$ {produtos[c+1]:>}\n')

print('------------------------------------------------------------------')

"""
    '<' - serve para alinhar caracteres à esquerda
    '>' - serve para alinhar caracteres à direita
    '58' - é a quantidade de posições/caracteres
    ':.<58' - contando o elemento 'produtos[c]', tudo será alinhado à esquerda e ocupará um o espaço de 58 caracteres, após o elemento 'produtos[c]', o espaço que resta será preenchido com pontos (.)
    ':<' - o elemento 'produtos[c+1]' será alinhado a direita

    esses comandos são sempre após os elementos que estão dentro das chaves quando se utiliza as f-strings

"""

-------------------------TABELA DE PREÇOS-------------------------

Arroz..................................................... R$ 5.53


Feijão.................................................... R$ 4.32


Macarrão.................................................. R$ 3.25


Farinha................................................... R$ 6.47


Carne..................................................... R$ 17.83


Frango.................................................... R$ 9.67

------------------------------------------------------------------


In [ ]:
# SOLUÇÃO DO GUANABARA

produtos = ('Arroz', 5.53, 'Feijão', 4.32, 'Macarrão', 3.25, 'Farinha', 6.47, 'Carne', 17.83, 'Frango', 9.67,)

print('-' * 40)
print(f'{"LISTAGEM DE PREÇOS":^40}') # :^40 - colocando "listagem de preços" centralizada e em 40 espaços
print('-' * 40)


for pos in range(0, len(produtos)): # pos = posição | len(produtos) = o tamanho da tupla
    if pos % 2 == 0: # serve para printar os produtos que estão nos índices pares, formatados dessa forma
        print(f'{produtos[pos]:.<30}', end='')
    else: # serve para printar os preços que estão nos índices ímpares, formatados dessa forma
        print(f' R$ {produtos[pos]:>5.2f}')


----------------------------------------
           LISTAGEM DE PREÇOS           
----------------------------------------
Arroz......................... R$  5.53
Feijão........................ R$  4.32
Macarrão...................... R$  3.25
Farinha....................... R$  6.47
Carne......................... R$ 17.83
Frango........................ R$  9.67


##Exercício Python 077: Crie um programa que tenha uma tupla com várias palavras (não usar acentos). Depois disso, você deve mostrar, para cada palavra, quais são as suas vogais.

In [ ]:

palavras = ('python', 'data', 'science', 'curso', 'video', 'tecnologia', 'dinheiro', 'felicidade') # não acentuar as palavras

for palavra in palavras: # esse for serve para iterar em cada palavra da tupla
    print(f'\nNa palavra {palavra.upper()} temos ', end='')
    for letra in palavra: # esse for serve para iterar sobre cada letra de acordo com cada palavra que esteja sendo lida pelo primeiro for
                          # (lembre que uma string é uma lista, ou seja, cada palavra em si é também uma lista)
        if letra.lower() in 'aeiou': # caso quiséssemos acentuar as palavras, teríamos que incluir as letras com acento em 'aeiou'
            print(letra, end=' ')



Na palavra PYTHON temos o 
Na palavra DATA temos a a 
Na palavra SCIENCE temos i e e 
Na palavra CURSO temos u o 
Na palavra VIDEO temos i e o 
Na palavra TECNOLOGIA temos e o o i a 
Na palavra DINHEIRO temos i e i o 
Na palavra FELICIDADE temos e i i a e 

# LISTAS - PARTE 1


## Exercício Python 078: Faça um programa que leia 5 valores numéricos e guarde-os em uma lista. No final, mostre qual foi o maior e o menor valor digitado e as suas respectivas posições na lista.


In [ ]:
# inicializando as variáveis
lista_num = []

for n in range(1, 6):
    num = float(input(f'Digite o {n}º número: '))
    lista_num.append(num)

# retornando os maior e o menor valor
maior = max(lista_num)
menor = min(lista_num)

# exibindo os resultados
print(lista_num)
print(f'O maior valor da lista é: {maior} '
      f'e sua posição (índice) na lista é: {lista_num.index(maior)}.') # nas listas usa-se a variável como parâmetro do método/função: a opção "lista_num.max()" NÃO FUNCIONA!
print(f'O menor valor da lista é: {menor} '
      f'e sua posição (índice) na lista é: {lista_num.index(menor)}.')

Digite o 1º número: 3
Digite o 2º número: 7
Digite o 3º número: 3
Digite o 4º número: 9
Digite o 5º número: 2
[3.0, 7.0, 3.0, 9.0, 2.0]
O maior valor da lista é: 9.0 e sua posição (índice) na lista é: 3.
O menor valor da lista é: 2.0 e sua posição (índice) na lista é: 4.


## Exercício Python 079: Crie um programa onde o usuário possa digitar vários valores numéricos e cadastre-os em uma lista. Caso o número já exista lá dentro, ele não será adicionado. No final, serão exibidos todos os valores únicos digitados, em ordem crescente.

In [ ]:
# inicializando as variáveis
lista_numeros = []

while True:
    num = float(input(f'Digite um número: '))
    if num not in lista_numeros:
        lista_numeros.append(num)
    else:
        num = float(input(f'Esse número já existe na lista! Digite outro número: '))

    resp = ''
    resp = input('Deseja continuar [S/N]?'
                 '\nOpção desejada: ').strip().upper()

    if resp not in 'SN':
        resp = input('Deseja continuar [S/N]?'
                     '\nOpção desejada: ').strip().upper()
    elif resp == 'N':
        break

"""lista_ordenada = lista_numeros.sort() - NÃO FUNCIONA - O python modifica a lista in-place e retorna None, ou seja, ele ordena a lista diretamente e não retorna uma nova lista ordenada.
                                            Assim, quando você faz lista_ordenada = lista_numeros.sort(), lista_ordenada fica com o valor None.""" # explicação do mano chatGPT

# ordenando a lista da forma correta, após o erro

lista_numeros.sort()

# exibindo os resultados
""" print(lista_numeros.sort()) - pelo mesmo motivo do 'lista_ordenada = lista_numeros.sort()', não é possível executar esse comando,
                                pois ele retornará NONE também."""

print(f'A lista ordenada dos números digitados é: {lista_numeros}')

Digite um número: 3
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 9
Deseja continuar [S/N]?
Opção desejada: 4
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 3
Esse número já existe na lista! Digite outro número: 5
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 8
Deseja continuar [S/N]?
Opção desejada: n
A lista ordenada dos números digitados é [3.0, 8.0, 9.0]


## Exercício Python 080: Crie um programa onde o usuário possa digitar cinco valores numéricos e cadastre-os em uma lista, já na posição correta de inserção (sem usar o sort()). No final, mostre a lista ordenada na tela.

In [ ]:
# exercício realizado assistindo ao vídeo de resolução, não consegue fazer sozinho

# criando a lista
lista = []

# lendo os 5 valores

""" para cada número digitado, só há três possibilidade: ou ele é o menor valor e ficará na primeira posição,
ou ele é o maior valor e ficará na última posição, ou ele ficará entre os valor do meio da lista"""

for n in range(0, 5):
    num = float(input(f'Digite o {n+1}º número: '))
    """if n == 0:
        lista.append(num) # como é o primeiro valor lido, posso usar o método append()
    elif num > lista[-1]: # comparando "num" com o último elemento da lista, poderia ser usado "num > lista[len(lista)-1]", pois também funionaria
        lista.append(num) # o método apeend() adiciona valores ao final da lista, como o último valor da lista deverá ser o maior valor, podemos utilizar esse método"""

    # simplificando a lógica acima
    if n == 0 or num > lista[-1]:
        lista.append(num)
        print('Adicionado ao final da lista...')
    else: # se não é o primeiro ou o último deverá ficar no meio da lista, portanto, temos que descobrir em qual posição ele ficará
        pos = 0 # posição inicial do vetor
        while pos < len(lista): # enquanto a posição inicial for menor que a última posição
            if num <= lista[pos]: # verificando se o número a ser inserido é menor ou igual ao número na posição "pos"
            # nesse caso teremos que inserir esse número em uma posição específica, portando, não podemos utilizar o método append(), devemos utilizar o método insert()
                lista.insert(pos, num) # inserindo o
                print(f'Adicionado na posição {pos} da lista...')
                break
            pos += 1

print('=-'*40)
print(f'\nOs valores digitados em ordem foram: {lista}.')


Digite o 1º número: 8
Adicionado ao final da lista...
Digite o 2º número: 1
Adicionado na posição 0 da lista...
Digite o 3º número: 7
Adicionado na posição 1 da lista...
Digite o 4º número: 10
Adicionado ao final da lista...
Digite o 5º número: 0
Adicionado na posição 0 da lista...
=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-

Os valores digitados em ordem foram: [0.0, 1.0, 7.0, 8.0, 10.0].


## Exercício Python 081: Crie um programa que vai ler vários números e colocar em uma lista. Depois disso, mostre:

1. Quantos números foram digitados.
2. A lista de valores, ordenada de forma decrescente.
3. Se o valor 5 foi digitado e está ou não na lista.

In [ ]:
# inicializando as variáveis
lista_numeros2 = []

# corpo do código
while True:
    num = float(input(f'Digite um número: '))
    if num not in lista_numeros2:
        lista_numeros2.append(num)
    else:
        num = float(input(f'Esse número já existe na lista! Digite outro número: ')) # eu quis aproveitar o código do exercício 79, para deixar apenas números únicos na lista

    resp = ''
    resp = input('Deseja continuar [S/N]?'
                 '\nOpção desejada: ').strip().upper()

    if resp not in 'SN':
        resp = input('Deseja continuar [S/N]?'
                     '\nOpção desejada: ').strip().upper()
    elif resp == 'N':
        break

# ordenando a lista de forma decrescente
lista_numeros2.sort(reverse=True)

# apresentando os resultados
print(f'Foram digitados {len(lista_numeros2)} números.')
print(f'Lista ordenada de forma decrescente: {lista_numeros2}.')

if 5 in lista_numeros2:
    print(f'O número 5 está na lista!')
else:
    print(f'O número 5 não está na lista!')


Digite um número: 2
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 4
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 6
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 8
Deseja continuar [S/N]?
Opção desejada: n
Foram digitados 4 números.
Lista ordenada de forma decrescente: [8.0, 6.0, 4.0, 2.0].
O número não está na lista!


## Exercício Python 082: Crie um programa que vai ler vários números e coloque em uma lista. Depois disso, crie duas listas extras que vão conter apenas os valores pares e os valores ímpares digitados, respectivamente. Ao final, mostre o conteúdo das três listas geradas.

In [ ]:
# inicializando as variáveis
lista_numeros = []
lista_pares = []
lista_impares = []

while True:
    num = float(input(f'Digite um número: '))
    if num not in lista_numeros:
        lista_numeros.append(num)
    else:
        num = float(input(f'Esse número já existe na lista! Digite outro número: '))

    if num % 2 == 0:
        lista_pares.append(num)
    else:
        lista_impares.append(num)

    resp = ''
    resp = input('Deseja continuar [S/N]?'
                 '\nOpção desejada: ').strip().upper()

    if resp not in 'SN':
        resp = input('Deseja continuar [S/N]?'
                     '\nOpção desejada: ').strip().upper()
    elif resp == 'N':
        break

# exibindo os resultados

print(f'A lista digitada foi: {lista_numeros}.')
print(f'Com base na lista original, a lista dos números pares é: {lista_pares}.')
print(f'Com base na lista original, a lista dos números ímpares é: {lista_impares}.')

Digite um número: 2
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 3
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 4
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 5
Deseja continuar [S/N]?
Opção desejada: 6
Deseja continuar [S/N]?
Opção desejada: s
Digite um número: 7
Deseja continuar [S/N]?
Opção desejada: n
A lista digitada foi: [2.0, 3.0, 4.0, 5.0, 7.0].
Com base na lista original, a lista dos números pares é: [2.0, 4.0].
Com base na lista original, a lista dos números ímpares é: [3.0, 5.0, 7.0].


## Exercício Python 083: Crie um programa onde o usuário digite uma expressão qualquer que use parênteses. Seu aplicativo deverá analisar se a expressão passada está com os parênteses abertos e fechados na ordem correta.

In [ ]:
expr = str(input('Digite a expressão: '))
pilha = []

# verificando os símbolos
for simb in expr:
    if simb == '(':
        pilha.append('(') # sempre que encontrar '(', ou seja, um parênteses aberto, o programa adicionará esse parênteses à pilha
    elif simb == ')':
        if len(pilha) > 0:
            """ sempre que encontrar ')', ou seja, um parênteses aberto, ele verificará se a pilha contém um parênteses aberto '(',
            se contiver, ele removerá o parêntes aberto da pilha, pois ele terá seu parênteses fechado, formando o par '()'.    """
            pilha.pop()
        else:
            pilha.append(')')
            break

if len(pilha) == 0:
    print('Sua expressão é válida!')
else:
    print('Sua expressão é inválida!')

Digite a expressão: (a+b)(a*(c+n))+((3-4)/3
Sua expressão é inválida!


# LISTAS - PARTE 2

## Exercício Python 084: Faça um programa que leia nome e peso de várias pessoas, guardando tudo em uma lista. No final, mostre:

1. Quantas pessoas foram cadastradas.
2. Uma listagem com as pessoas mais pesadas.
3. Uma listagem com as pessoas mais leves.

In [10]:
# criando as listas
pessoas = list()
individuo = list()
totPessoas = 0
maisPesadas = list()
maisLeves = list()


while True:
    # entrada de dados
    individuo.append(str(input('Digite o nome da pessoa: ')))
    individuo.append(float(input(f'Digite o peso do(a) {individuo[0]}: ')))
    pessoas.append(individuo[:]) # lembre-se que ao utilizar "individuo[:]" significa que será uma cópia da lista individuo
    individuo.clear()
    totPessoas += 1

    resposta = ''
    resposta = input('Deseja continuar [S/N]?'
                 '\nOpção desejada: ').strip().upper()

    # verificando se o usuário deseja continuar cadastrando, caso não, o loop será interrompido
    if resposta not in 'SN':
        resposta = input('Deseja continuar [S/N]?'
                     '\nOpção desejada: ').strip().upper()
    elif resposta == 'N':
        break

# verificando os mais pesados
for p in pessoas: # para cada pessoa na lista
    if p[1] >= 90: # verificando apenas a idade de cada pessoa na lista
        maisPesadas.append(p[0]) # printando o nome dessa pessoa com o peso maior ou igual a 90 kg

    # verificando os mais leves
for p in pessoas: # para cada pessoa na lista
    if p[1] <= 65: # verificando apenas a idade de cada pessoa na lista
        maisLeves.append(p[0]) # printando o nome dessa pessoa com o peso maior ou igual a 90 kg


print(pessoas)
print(f'Foram cadastradas {totPessoas} pessoas.')
print(f'As pessoas mais leves são: {maisLeves}.')
print(f'As pessoas mais pesadas são: {maisPesadas}.')

Digite o nome da pessoa: Pedro
Digite o peso do(a) Pedro: 110
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: João
Digite o peso do(a) João: 80
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: Márcia
Digite o peso do(a) Márcia: 55
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: Joana
Digite o peso do(a) Joana: 75
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: Cleiton
Digite o peso do(a) Cleiton: 90
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: Cláudia
Digite o peso do(a) Cláudia: 60
Deseja continuar [S/N]?
Opção desejada: Milena
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome da pessoa: Milena
Digite o peso do(a) Milena: 85
Deseja continuar [S/N]?
Opção desejada: n
[['Pedro', 110.0], ['João', 80.0], ['Márcia', 55.0], ['Joana', 75.0], ['Cleiton', 90.0], ['Cláudia', 60.0], ['Milena', 85.0]]
Foram cadastradas 7 pessoas.
As pessoas mais leves são: ['Márcia', 'Cláudia'].
As pessoas mais 

## Exercício Python 085: Crie um programa onde o usuário possa digitar sete valores numéricos e cadastre-os em uma lista única que mantenha separados os valores pares e ímpares. No final, mostre os valores pares e ímpares em ordem crescente.

In [22]:
# declarando as variáveis
numeros = list()
pares = list()
impares = list()

for n in range(1,8):
    num = int(input(f'Digite o {n}º número: '))

    # verificando se é par ou ímpar
    if num % 2 == 0:
        pares.append(num)
    else:
        impares.append(num)

pares.sort()
impares.sort()
numeros.append(pares)
numeros.append(impares)
print(numeros)
print(pares)
print(impares)


Digite o 1º número: 5
Digite o 2º número: 2
Digite o 3º número: 6
Digite o 4º número: 3
Digite o 5º número: 9
Digite o 6º número: 4
Digite o 7º número: 7
[[2, 4, 6], [3, 5, 7, 9]]
[2, 4, 6]
[3, 5, 7, 9]


## Exercício Python 086: Crie um programa que declare uma matriz de dimensão 3x3 e preencha com valores lidos pelo teclado. No final, mostre a matriz na tela, com a formatação correta.

In [28]:
# declarando as variáveis
matriz = list()
pLinha = list()
sLinha = list()
tLinha = list()

for n in range(1,10):
    num = int(input(f'Digite o {n}º elemento: '))

    if n < 4:
        pLinha.append(num)
    elif n < 7:
        sLinha.append(num)
    elif n < 10:
        tLinha.append(num)

matriz.append(pLinha)
matriz.append(sLinha)
matriz.append(tLinha)

for l in matriz:
    print(l, end='\n')

Digite o 1º elemento: 1
Digite o 2º elemento: 2
Digite o 3º elemento: 3
Digite o 4º elemento: 4
Digite o 5º elemento: 5
Digite o 6º elemento: 6
Digite o 7º elemento: 7
Digite o 8º elemento: 8
Digite o 9º elemento: 9
[1, 2, 3]
[4, 5, 6]
[7, 8, 9]


## Exercício Python 087: Aprimore o desafio anterior, mostrando no final:

1. A soma de todos os valores pares digitados.
2. A soma dos valores da terceira coluna.
3. O maior valor da segunda linha.

In [45]:
# declarando as variáveis
matriz = list()
pLinha = list()
sLinha = list()
tLinha = list()
somaPar = 0
somaTer = 0


for n in range(1,10):
    num = int(input(f'Digite o {n}º elemento: '))

    if n < 4:
        pLinha.append(num)
    elif n < 7:
        sLinha.append(num)
    elif n < 10:
        tLinha.append(num)

matriz.append(pLinha)
matriz.append(sLinha)
matriz.append(tLinha)

print('\n---------- FORMA MATRICIAL ----------')
for linha in matriz:
    print(linha, end='\n')

print('\n------------ RESULTADOS -------------')
# somando os elementos pares da matriz:
for linha in matriz:
    for numero in linha:
        if numero % 2 == 0:
            somaPar += numero
print(f'A soma dos elementos pares da matriz é: {somaPar}.')

# somando os elementos que se encontram na terceira coluna, ou seja, os elementos de índice [2] de cada lista que corresponde a linha
for linha in matriz:
        somaTer += linha[:][2]
print(f'A soma dos terceiros elementos da matriz é: {somaTer}.')

# maior valor da segunda linha
maior = max(matriz[1])
print(f'O maior valor da segunda linha da matriz é: {maior}.')

Digite o 1º elemento: 1
Digite o 2º elemento: 2
Digite o 3º elemento: 3
Digite o 4º elemento: 4
Digite o 5º elemento: 5
Digite o 6º elemento: 6
Digite o 7º elemento: 7
Digite o 8º elemento: 8
Digite o 9º elemento: 9

---------- FORMA MATRICIAL ----------
[1, 2, 3]
[4, 5, 6]
[7, 8, 9]

------------ RESULTADOS -------------
A soma dos elementos pares da matriz é: 20.
A soma dos terceiros elementos da matriz é: 18.
O maior valor da segunda linha da matriz é: 6.


## Exercício Python 088: Faça um programa que ajude um jogador da MEGA SENA a criar palpites.O programa vai perguntar quantos jogos serão gerados e vai sortear 6 números entre 1 e 60 para cada jogo, cadastrando tudo em uma lista composta.

## Exercício Python 089: Crie um programa que leia nome e duas notas de vários alunos e guarde tudo em uma lista composta. No final, mostre um boletim contendo a média de cada um e permita que o usuário possa mostrar as notas de cada aluno individualmente.

In [53]:
# criando as listas
alunos = list()
aluno = list()
media = 0

while True:
    # entrada de dados
    aluno.append(str(input('Digite o nome do(a) aluno(a): ')))
    aluno.append(float(input(f'Digite a 1ª nota do(a) {aluno[0]}: ')))
    aluno.append(float(input(f'Digite a 2ª nota do(a) {aluno[0]}: ')))
    alunos.append(aluno[:]) # lembre-se que ao utilizar "aluno[:]" significa que será uma cópia da lista aluno
    aluno.clear()

    resposta = ''
    resposta = input('Deseja continuar [S/N]?'
                 '\nOpção desejada: ').strip().upper()

    # verificando se o usuário deseja continuar cadastrando, caso não, o loop será interrompido
    if resposta not in 'SN':
        resposta = input('Deseja continuar [S/N]?'
                     '\nOpção desejada: ').strip().upper()
    elif resposta == 'N':
        break

print(alunos, end='\n')

for i in alunos:
    media = (i[1] + i[2])/2
    print(f'A média do(a) aluno {i[0]} é: {media}.')


Digite o nome do(a) aluno(a): joao
Digite a 1ª nota do(a) joao: 5
Digite a 2ª nota do(a) joao: 7
Deseja continuar [S/N]?
Opção desejada: s
Digite o nome do(a) aluno(a): julia
Digite a 1ª nota do(a) julia: 7
Digite a 2ª nota do(a) julia: 8
Deseja continuar [S/N]?
Opção desejada: n
[['joao', 5.0, 7.0], ['julia', 7.0, 8.0]]
A média do(a) aluno joao é: 6.0.
A média do(a) aluno julia é: 7.5.


# DICIONÁRIOS

## Exercício Python 090: Faça um programa que leia nome e média de um aluno, guardando também a situação em um dicionário. No final, mostre o conteúdo da estrutura na tela.

## Exercício Python 091: Crie um programa onde 4 jogadores joguem um dado e tenham resultados aleatórios. Guarde esses resultados em um dicionário em Python. No final, coloque esse dicionário em ordem, sabendo que o vencedor tirou o maior número no dado.

## Exercício Python 092: Crie um programa que leia nome, ano de nascimento e carteira de trabalho e cadastre-o (com idade) em um dicionário. Se por acaso a CTPS for diferente de ZERO, o dicionário receberá também o ano de contratação e o salário. Calcule e acrescente, além da idade, com quantos anos a pessoa vai se aposentar.

## Exercício Python 093: Crie um programa que gerencie o aproveitamento de um jogador de futebol. O programa vai ler o nome do jogador e quantas partidas ele jogou. Depois vai ler a quantidade de gols feitos em cada partida. No final, tudo isso será guardado em um dicionário, incluindo o total de gols feitos durante o campeonato.

## Exercício Python 094: Crie um programa que leia nome, sexo e idade de várias pessoas, guardando os dados de cada pessoa em um dicionário e todos os dicionários em uma lista. No final, mostre:

1. Quantas pessoas foram cadastradas
2. A média de idade
3. Uma lista com as mulheres
4. Uma lista de pessoas com idade acima da média

## Exercício Python 095: Aprimore o desafio 93 para que ele funcione com vários jogadores, incluindo um sistema de visualização de detalhes do aproveitamento de cada jogador.

# FUNÇÕES - PARTE 1

## Exercício Python 096: Faça um programa que tenha uma função chamada área(), que receba as dimensões de um terreno retangular (largura e comprimento) e mostre a área do terreno.

## Exercício Python 097: Faça um programa que tenha uma função chamada escreva(), que receba um texto qualquer como parâmetro e mostre uma mensagem com tamanho adaptável.

#### Ex:
#### escreva('Olá, Mundo!')
#### Saída:
#### Olá, Mundo!


## Exercício Python 098: Faça um programa que tenha uma função chamada contador(), que receba três parâmetros: início, fim e passo. Seu programa tem que realizar três contagens através da função criada:

1. de 1 até 10, de 1 em 1
2. de 10 até 0, de 2 em 2
3. uma contagem personalizada

## Exercício Python 099: Faça um programa que tenha uma função chamada maior(), que receba vários parâmetros com valores inteiros. Seu programa tem que analisar todos os valores e dizer qual deles é o maior.

## Exercício Python 100: Faça um programa que tenha uma lista chamada números e duas funções chamadas sorteia() e somaPar(). A primeira função vai sortear 5 números e vai colocá-los dentro da lista e a segunda função vai mostrar a soma entre todos os valores pares sorteados pela função anterior.

# FUNÇÕES - PARTE 2

## Exercício Python 101: Crie um programa que tenha uma função chamada voto() que vai receber como parâmetro o ano de nascimento de uma pessoa, retornando um valor literal indicando se uma pessoa tem voto NEGADO, OPCIONAL e OBRIGATÓRIO nas eleições.

## Exercício Python 102: Crie um programa que tenha uma função fatorial() que receba dois parâmetros: o primeiro que indique o número a calcular e outro chamado show, que será um valor lógico (opcional) indicando se será mostrado ou não na tela o processo de cálculo do fatorial.

## Exercício Python 103: Faça um programa que tenha uma função chamada ficha(), que receba dois parâmetros opcionais: o nome de um jogador e quantos gols ele marcou. O programa deverá ser capaz de mostrar a ficha do jogador, mesmo que algum dado não tenha sido informado corretamente.

## Exercício Python 104: Crie um programa que tenha a função leiaInt(), que vai funcionar de forma semelhante 'a função input() do Python, só que fazendo a validação para aceitar apenas um valor numérico.

> Ex: n = leiaInt('Digite um n: ')



## Exercício Python 105: Faça um programa que tenha uma função notas() que pode receber várias notas de alunos e vai retornar um dicionário com as seguintes informações:

1. Quantidade de notas
2. A maior nota
3. A menor nota
4. A média da turma
5. A situação (opcional)

> Adicione também as docstrings dessa função para consulta pelo desenvolvedor.



## Exercício Python 106: Faça um mini-sistema que utilize o Interactive Help do Python. O usuário vai digitar o comando e o manual vai aparecer. Quando o usuário digitar a palavra 'FIM', o programa se encerrará. Importante: use cores.

# MÓDULOS E PACOTES

## Exercício Python 107: Crie um módulo chamado moeda.py que tenha as funções incorporadas aumentar(), diminuir(), dobro() e metade(). Faça também um programa que importe esse módulo e use algumas dessas funções.

## Exercício Python 108: Adapte o código do desafio 107, criando uma função adicional chamada moeda() que consiga mostrar os números como um valor monetário formatado.

## Exercício Python 109: Modifique as funções que form criadas no desafio 107 para que elas aceitem um parâmetro a mais, informando se o valor retornado por elas vai ser ou não formatado pela função moeda(), desenvolvida no desafio 108.

## Exercício 110 – Reduzindo ainda mais seu programa


## Exercício Python 111: Crie um pacote chamado utilidadesCeV que tenha dois módulos internos chamados moeda e dado. Transfira todas as funções utilizadas nos desafios 107, 108 e 109 para o primeiro pacote e mantenha tudo funcionando.

## Exercício Python 112: Dentro do pacote utilidadesCeV que criamos no desafio 111, temos um módulo chamado dado. Crie uma função chamada leiaDinheiro() que seja capaz de funcionar como a função imputa(), mas com uma validação de dados para aceitar apenas valores que seja monetários.

# TRATAMENTO DE ERROS E EXCEÇÕES

## Exercício Python 113: Reescreva a função leiaInt() que fizemos no desafio 104, incluindo agora a possibilidade da digitação de um número de tipo inválido. Aproveite e crie também uma função leiaFloat() com a mesma funcionalidade.

## Exercício Python 114: Crie um código em Python que teste se o site pudim está acessível pelo computador usado.